simpeg/directives/_regularization.py

```diff
@@ -0,0 +1,494 @@
+        self.chifact_start: float = chifact_start
+        self.chifact_target: float = chifact_target
+        self.irls_cooling_factor: float = irls_cooling_factor
+        self.f_min_change: float = f_min_change
+        self.max_irls_iterations: int = max_irls_iterations
+        self.misfit_tolerance: float = misfit_tolerance
+        self.percentile: float = percentile
+
+        super().__init__(
+            verbose=verbose,
+            **kwargs,
+        )
+
+    @property
+    def metrics(self) -> IRLSMetrics:
+        """Various metrics used by the IRLS algorithm."""
+        if self._metrics is None:
+            self._metrics = IRLSMetrics()
+        return self._metrics
+
+    @property
+    def max_irls_iterations(self) -> int:
+        """Maximum irls iterations."""
+        return self._max_irls_iterations
+
+    @max_irls_iterations.setter
+    def max_irls_iterations(self, value):
+        self._max_irls_iterations = validate_integer(
+            "max_irls_iterations", value, min_val=0
+        )
+
+    @property
+    def misfit_tolerance(self) -> float:
+        """Tolerance on deviation from the target chi factor, as a fractional percent."""
+        return self._misfit_tolerance
+
+    @misfit_tolerance.setter
+    def misfit_tolerance(self, value):
+        self._misfit_tolerance = validate_float("misfit_tolerance", value, min_val=0)
+
+    @property
+    def percentile(self) -> float:
+        """Tolerance on deviation from the target chi factor, as a fractional percent."""
+        return self._percentile
+
+    @percentile.setter
+    def percentile(self, value):
+        self._percentile = validate_float(
+            "percentile", value, min_val=0.0, max_val=100.0
+        )
+
+    @property
+    def chifact_start(self) -> float:
+        """Target chi factor to start the IRLS process."""
+        return self._chifact_start
+
+    @chifact_start.setter
+    def chifact_start(self, value):
+        self._chifact_start = validate_float(
+            "chifact_start", value, min_val=0, inclusive_min=False
+        )
+
+    @property
+    def chifact_target(self) -> float:
+        """Targer chi factor to maintain during the IRLS process."""
+        return self._chifact_target
+
+    @chifact_target.setter
+    def chifact_target(self, value):
+        self._chifact_target = validate_float(
+            "chifact_target", value, min_val=0, inclusive_min=False
+        )
+
+    @property
+    def cooling_factor(self):
+        """Beta is divided by this value every `cooling_rate` iterations.
+
+        Returns
+        -------
+        float
+        """
+        return self._cooling_factor
+
+    @cooling_factor.setter
+    def cooling_factor(self, value):
+        self._cooling_factor = validate_float(
+            "cooling_factor", value, min_val=0.0, inclusive_min=False
+        )
+
+    @property
+    def cooling_rate(self):
+        """Cool after this number of iterations.
```

Comment:
Minor update to make this property more clear.
```suggestion
        """Cool beta after this number of iterations.
```